solr/packaging/test/test_placement_plugin.bats

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bats
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load bats_helper
+
+setup() {
+  common_clean_setup
+}
+
+teardown() {
+  # save a snapshot of SOLR_HOME for failed tests
+  save_home_on_failure
+
+  delete_all_collections
+  solr stop -all >/dev/null 2>&1
+}
+
+@test "Affinity placement plugin using sysprop" {
+  run solr start -c -Dsolr.defaultPlacementPlugin=affinity
+  sleep 3
+  run solr create_collection -c COLL_NAME
+  sleep 3
+  run cat ${SOLR_HOME}/logs/solr.log
```

Comment:
Is there a smarter way to check Solr's log than waiting a few secs and cat'ing it? :) I don't like sleep in tests

Reply:
He might be busy but @madrob is probably the best person to answer this....

Reply:
might be worth looking at adding https://github.com/bats-core/bats-file - which has https://github.com/bats-core/bats-file#assert_file_contains

Reply:
An example of changes required to load the bats stuff is here: https://github.com/apache/solr/pull/1091

Reply:
Having to sleep is a symptom of us not supporting a collection ready API, but bats-file should do the cat part for you.

Reply:
Nice, I struggled a bit with installing the correct version of bats-file, but once I settled on the latest versions from the https://github.com/bats-core repo, all was good.

Removed sleep after start with an undocumented solr command which will retry for a configurable time:
```bash
solr assert -c http://localhost:8983/solr -t 3000
```

Reply:
@risdenk The `bats` package in NPM is the correct one, but `ztombol/bats-support#v0.2.0`, `ztombol/bats-assert#v0.3.0` are old versions from old repo, and the `ztombol/bats-file#v0.2.0` does not have the `assert_file_contains` method. Thus I pinned all of these on the latest from `bats-core` github repo.

Reply:
Nice I couldn't figure out why bats-core stuff wouldn't work. Guess removing bats-support was all that is necessary